### DISCUSSION TITLE
Complete conditional cycle loops with ENDxx

### DISCUSSION THREAD

**Discussion by lgiammattei** [codefori/vscode-ibmi#1246] [General]
I apologize if this has already been discussed, I did a quick search before opening a new thread but was unable to find an answer.
While I'm editing a rpgle or sqlrpgle source, if I code a do, for loop, or a Select or if check, the statement with the corresponding ENDxx is not completed. Is there something I need to check in the settings? Or do I have to go and create some specific code snippets?

**Comment by worksofliam** [comment]
Hi!

Ensure you have the RPGLE extension installed. This extension adds the snippets for block statements, and adds the end that you are looking for. It works for 'if', 'select', 'for', etc

**Reply by lgiammattei** [reply]
Hi Liam thanks for your reply, i have the RPGLE extension installed and enabled (version 0.17.6), I installed the code for ibm i pack.

**Reply by lgiammattei** [reply]
Okay it works. I understood where I was wrong.
Consider that I come from using RDi, I was used in writing "If;" "dow;" etc and find the relative "Endxx" autocompleted.
Working with your extensions in VSC I have to select the completion hint while typing "Sele" for example, and I will find the related "When" "Other" and "ENDSL" completed on the source file. If I complete the line statement by writing "Select;" evidently the need to write the completion of the conditional loop is not intercepted.
